ui/app/actions.js

```diff
@@ -669,12 +671,24 @@ function markAccountsFound() {
 // config
 //
 
-function setRpcTarget (newRpc) {
-  log.debug(`background.setRpcTarget`)
-  background.setRpcTarget(newRpc)
+// default rpc target refers to localhost:8545 in this instance.
+function setDefaultRpcTarget (rpcList) {
+  log.debug(`background.setDefaultRpcTarget`)
+  background.setDefaultRpc()
   return {
     type: actions.SET_RPC_TARGET,
-    value: newRpc,
+    value: 'http://localhost:8545',
+  }
+}
+
+function setRpcTarget (newRpc) {
+  return (dispatch) => {
+    log.debug(`background.setRpcTarget`)
+    background.setCustomRpc(newRpc)
+    return {
+      type: actions.SET_RPC_TARGET,
+      value: newRpc,
+    }
```

Comment:
I'm not sure you can return an action from a `dispatch` callback.  Can you?  I know you can `dispatch({ type, value })`.

I'm still somewhat inclined that you should be using callbacks on your calls to the `background`, and handling the `error` correctly.  It really isn't that hard, there are good examples all throughout this file, [like this one](https://github.com/MetaMask/metamask-plugin/blob/master/ui/app/actions.js#L216).

I *almost* was going to say you don't need a callback here, since you're triggering a reload in that action, but if there's a network error between the `popup` and `background`, then there could be an `error` in the callback that doesn't have to do with the background process, and we should be able to represent those types of errors.